module_utils/network_lsr/argument_validator.py

```diff
@@ -271,6 +271,7 @@ def _validate_impl(self, value, name):
                 except Exception:
                     table = value
         except Exception:
```

Comment:
@thom311 @liangwen12year what exceptions are we excepting here? Is it a ValueError from line 267? It seems like it could be excepted there specifically.

Reply:
I don't think any exception can be raised here. The outer try/except should be dropped.

Reply:
Ah, I see now. So `table = value` would be better in line 267, wouldn't it?

Reply:
If the value is already the int type, then the function `int()` seems to me is not needed.

Reply:
`isinstance()` does not check whether it's the same type, but of the type or a subtype.

the point of the code is of course to ensure that the result is really an int, and not a subtype.

"not needed", is if we never call the function with a subtype of int (`bool` is also a subtype of `int`, but already handled above).

I would leave it.

Reply:
What's the problem if it is a subclass? Regarding `bool`, the check is there to avoid accepting boolean values but other subclasses are not rejected, so it is not clear to me why this would be useful.

Reply:
the idea is that the validation+normalization returns ... normalized values. That is, `int` and not a subclass of an `int` that was passed as input to the validation.

Anyway, since nobody is actually subclassing `int` (aside `bool`, which is handled already), it doesn't have any effect.

Reply:
Normalization can also mean to normalize it to `int` or a subclass of it.